src/webgpu/api/validation/image_copy/buffer_texture_copies.spec.ts

```diff
@@ -304,3 +306,46 @@ g.test('depth_stencil_format,copy_buffer_offset')
       unreachable();
     }
   });
+
+g.test('texture_buffer_usages')
+  .desc(
+    `
+  Tests calling copyTextureToBuffer or copyBufferToTexture with the texture and the buffer missed
+  COPY_SRC, COPY_DST usage respectively.
+    - texture and buffer {with, without} COPY_SRC and COPY_DST usage.
+  `
+  )
+  .paramsSubcasesOnly(u =>
+    u //
+      .combine('textureUsage', kTextureUsages)
+      .combine('bufferUsage', kBufferUsages)
```

Comment:
nit: Testing all combinations is maybe slightly excessive. How about, add a filter after this that filters out cases where both the source and destination are invalid? With a little refactor it shouldn't be too overly complex. Hard to explain so here's what I'm thinking:

```ts
const kRequiredTextureUsage = {
  CopyT2B: GPUConst.TextureUsage.COPY_SRC,
  CopyB2T: GPUConst.TextureUsage.COPY_DST,
};
const kRequiredBufferUsage = {
  CopyB2T: GPUConst.BufferUsage.COPY_SRC,
  CopyT2B: GPUConst.BufferUsage.COPY_DST,
};
```

```ts
  .params(u =>
    u //
      .combine('copyType', ['CopyB2T', 'CopyT2B'] as const)
      .beginSubcases()
      .combine('textureUsage', kTextureUsages)
      .expand('_textureUsageValid', p => [p.textureUsage === kRequiredTextureUsage[p.copyType]])
      .combine('bufferUsage', kBufferUsages)
      .expand('_bufferUsageValid', p => [p.bufferUsage === kRequiredBufferUsage[p.copyType]])
      .filter(p => p._textureUsageValid || p._bufferUsageValid)
  )
```

```ts
    const isSuccess = _textureUsageValid && _bufferUsageValid;
    if (copyType === 'CopyB2T') {
      t.testCopyTextureToBuffer({ texture }, { buffer }, textureSize, isSuccess);
    } else if (copyType === 'CopyT2B') {
      t.testCopyBufferToTexture({ buffer }, { texture }, textureSize, isSuccess);
    }
```

Reply:
Thanks for suggesting a lighter way! It works.